### ISSUE TITLE
"Navigate to Member" spelling correction fails for numbers

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#18309]
**Version Used**: VS 2017 (15.0.0+26228.9)

**Steps to Reproduce**:

1. Open Roslyn.sln
2. Press <kbd>Ctrl</kbd>+<kbd>T</kbd>
3. Type `m TestDesconstruction4`

**Expected Behavior**:

The `TestDeconstruction4()` method is selected.

**Actual Behavior**:

The `TestDeconstruction()` method is selected.

**Additional Steps**:

If you type `m TestDesconstruction44` (with a second `4` at the end), the `TestDeconstruction4()` method is now selected. It seems the first number is being ignored when making a spelling correction to the input?

**Comment by CyrusNajmabadi** [comment]
Works without the 'm':

![image](https://cloud.githubusercontent.com/assets/4564579/24524364/1c0dbd98-154b-11e7-987c-be05c87e452b.png)

And with the 'm' for me:

![image](https://cloud.githubusercontent.com/assets/4564579/24524375/248893f8-154b-11e7-83ca-a56ff4baf416.png)

**Comment by CyrusNajmabadi** [comment]
Can you provide more info to repro this?

**Comment by CyrusNajmabadi** [comment]
Note: TestDesconstruction4 is mispelled.  (note the additional "s" in "Desc").   As such, only fuzzy matching finds the results.  Because both TestDeconstruction and TestDeconstruction4 are fuzzy matches, they are then sorted *by the platform such that TestDeconstruction comes first.

Ideally TestDeconstruction4 would come first as it's edit distance is shorter than TestDeconstruction.

**Comment by CyrusNajmabadi** [comment]
Tagging @grabalon as well as he's looking at the core platform component of this, and we'd certainly like this situation to be improved.

**Comment by Pilchie** [comment]
@CyrusNajmabadi couldn't we address this on the IDE side by modifying the secondary sort string
(Not that we necessarily want to fix it in that way).

**Comment by CyrusNajmabadi** [comment]
No.  Secondary sort string takes effect *if* the display names are the same.  See:

http://index/?leftProject=Microsoft.VisualStudio.Shell.UI.Internal&leftSymbol=ekl1fnvyyexu&file=Packages%5CNavigateTo%5CNavigateToItemResult.cs&line=382

and: 

http://index/?leftProject=Microsoft.VisualStudio.Shell.UI.Internal&leftSymbol=ekl1fnvyyexu&file=Packages%5CNavigateTo%5CNavigateToItemResult.cs&line=437

Basically, only if everything else matches do we use secondary sort.  It's not a means to sort prior to the other checks :-/

**Comment by Pilchie** [comment]
Okay - so resolving as external - @grabalon - do you have a link to the work item tracking this in the platform?
